scripts/train_bert.py

```diff
@@ -0,0 +1,60 @@
+from argparse import ArgumentParser
+
+import pytorch_lightning as pl
+import wandb
+from pytorch_lightning import Trainer
+from pytorch_lightning.loggers import WandbLogger
+
+from thegreatknowledgeheist.data import get_dataloaders
+from thegreatknowledgeheist.io import load_yaml
+from thegreatknowledgeheist.models.bert import (
+    AcronymIdentificationBert,
+    AmazonPolarityBert,
+    SwagBert,
+)
+
+GET_MODEL = {
+    "amazon_polarity": AmazonPolarityBert,
+    "acronym_identification": AcronymIdentificationBert,
+    "swag": SwagBert,
+}
+
+
+def train_model(model, dataloaders, config):
+    checkpoint_callback = pl.callbacks.ModelCheckpoint(
+        monitor="val_loss",
+        dirpath=f"{config['outputs_path']}/model_checkpoints",
+        filename=config["task"] + "-model-{epoch:02d}-{val_accuracy:.2f}",
+        save_top_k=1,
+        mode="min",
+    )
+
+    trainer = Trainer(
+        logger=WandbLogger(
+            save_dir=f"{config['outputs_path']}/logs",
+            project="TheGreatKnowledgeTransferAcronym",
+            entity="mma",
+        ),
+        gpus=config["gpus"],
+        max_epochs=config["max_epochs"],
+        callbacks=[checkpoint_callback],
+    )
+
+    trainer.fit(model, dataloaders["train"], dataloaders["val"])
+    wandb.finish()
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.add_argument("config_path", type=str, help="Path to config yaml")
+    args = parser.parse_args()
+    config = load_yaml(args.config_path)
+    dataloaders = get_dataloaders(
+        dataset_name=config["task"],
+        path_to_dataset=f"{config['dataset_path']}/{config['task']}",
```

Comment:
Zmieńmy to na path_to_dataset=config['dataset_path'], albo zmieńmy skrypt download_and_prepare_dataset żeby on zawsze zapisywał do podkatalogu z nazwą taska.

Obecnie żeby to zadziałało, to w download_and_prepare_dataset trzeba podać data/swag, a tutaj po prostu data. Więc przed użyciem skryptu do pobierania danych trzeba przeczytać kod tego skryptu, żeby wiedzieć jak nazwać podkatalogi do których pobieramy dane.

Reply:
poprawiam